Library/Homebrew/utils/livecheck_formula.rb

```diff
@@ -27,4 +27,68 @@ def parse_livecheck_response(response)
       livecheck_version: latest_version,
     }
   end
+
+  # A class that can be used to mimic the parsed command-lione arguments used
+  # by livecheck. Necessary parameter (`args`) for `Livecheck.livecheck_formulae`.
+  #
+  # @api private
+  class LivecheckArgs
```

Comment:
Feels like this should be part of `audit`?

Reply:
We might need to use it here as well. Currently the `LivecheckFormula` module uses the output of the `dev-cmd`, but with this class it can now call the `livecheck_formulae` method internally. That's why I moved it here (and it also seems like a 'utility' class to use anywhere we want to call `livecheck_formulae`).

If it has to be in audit, I'll move it.

Reply:
Doesn't have to move. It does make me wonder if it wouldn't be better to change the interface for `Homebrew::Livecheck.livecheck_formulae` to take the same arguments as these instead and `LivecheckFormula` as a class should go away entirely. If you're feeling even more ambitious it'd be nice to have the `livecheck`-calling logic extracted from the `Repology` module because it feels a bit weird to be there.

None of the `LivecheckFormula` or `Repology` stuff needs to (or should really) be in this PR, though!

Reply:
I've worked around this using `OpenStruct`: if the arguments are passed as a hash, I use `OpenStruct` to make them accessible using methods. So we don't need the `LivecheckArgs` class any more.

I've made note of the `LivecheckFormula` and `Repology` suggestions, I'll open up a PR soon after this one's done 👍🏼